cpp/src/hash/helper_functions.cuh

```diff
@@ -48,41 +48,11 @@ __inline__ __device__ int16_t atomicCAS(int16_t* address, int16_t compare, int16
   return (int16_t)atomicCAS(base_address, int_comp, int_val);
 }
 
-__inline__ __device__ int64_t atomicCAS(int64_t* address, int64_t compare, int64_t val)
-{
-  return (int64_t)atomicCAS((unsigned long long*)address, (unsigned long long)compare, (unsigned long long)val);
-}
-
-__inline__ __device__ uint64_t atomicCAS(uint64_t* address, uint64_t compare, uint64_t val)
-{
-  return (uint64_t)atomicCAS((unsigned long long*)address, (unsigned long long)compare, (unsigned long long)val);
-}
-
 __inline__ __device__ long long int atomicCAS(long long int* address, long long int compare, long long int val)
```

Comment:
Wait, why are there still atomicCAS functions in this file? Shouldn't they all have been incorporated in cpp/src/utilities/device_atomics.cuh?

Reply:
Oh yeah, about that, I found that my compiler complained about the equivalence of `long long` and `long long int` (it thinks they're different) and the overload in `device_atomics.cuh` uses `int64_t` which was set to `long long` in my system headers. I would suggest making the current overloads templated like so 
```c++
template <typename T>
__forceinline__ __device__
T atomicCAS(T* address, T compare, T val)
{
    return cudf::detail::typesAtomicCASImpl<T, sizeof(T)>()(address, compare, val);
}
```
but I didn't include them here because it will be a pain for @kovaltan while merging #1308.
As for the preceding `atomicCAS(int8_t)`, it can be removed once #1308 is merged.

Reply:
My question is more about why there are atomic functions in two places. Why can't they all go in cpp/src/utilities/device_atomics.cuh?

Reply:
Also, to avoid losing your suggestions, please file an issue.

Reply:
>Why can't they all go in cpp/src/utilities/device_atomics.cuh?

They can and they should. I'll do that if #1308 gets merged first. Or I'll let @kovaltan know because if this PR gets merged first, he'll have a ODR violation compilation issue.

Reply:
PR #1308 just got merged. Please try it.

Reply:
Great, 👏🏼 now **I** have the ODR violation.